src/Network/Haskoin/Store/Data.hs

```diff
@@ -887,6 +1001,32 @@ instance BinSerial XPubSummary where
         putWord64be (fromIntegral $ length ts)
         forM_ ts $ binSerial net
 
+    binDeserial net = do
+      r <- get
+      c <- get
+      z <- get
+      ext <- get
+      ch <- get
+      si <- toInteger <$> getWord64be
+      ps <- deserPaths net si
```

Comment:
I don't like custom instances like this. Convert the `HashMap Address [KeyIndex]` into an `[(Address, [KeyIndex])]` list by means of the `toList` function. Then use `map (first (runPut . binSerial net))` on that list to get `[(ByteString, [KeyIndex])]`, then just `put` that. There should be a default `Serialize` instance for it. To deserialize you need to reverse this.